Breakthrough-Energy/PowerSimData
# data: Eastern v41 data fixes

This data pull fixes three main issues:

1. Generator capacities for the dummy generators are restored to their intended capacities of 1 MW.
2. Pmin data is restored for all generators (even though these values are ignored in REISE, it's good to have them in our dataset)
3. Gencost curves are fixed to match the scaling from v41.

How to test that these data modifications work as intended:
1. Pull the latest version of `matreader`
2. Check out this branch
3. Rebase this branch onto `matreader`
4. `pip install .`
5. Create a one-day-long scenario.
```
from powersimdata.scenario.scenario import Scenario
scenario = Scenario('')
scenario.state.set_builder(['Eastern'])
scenario.state.builder.set_name('test', 'whatever')
scenario.state.builder.set_time('2016-01-01 00:00:00', '2016-01-01 23:00:00', '24H')
scenario.state.builder.set_base_profile('demand', 'v6')  # set demand
scenario.state.builder.set_base_profile('hydro', 'v4')  # set hydro
scenario.state.builder.set_base_profile('solar', 'v4')  # set solar
scenario.state.builder.set_base_profile('wind', 'v5')  # set wind
scenario.state.create_scenario()
scenario.state.prepare_simulation_input()
```
6. Check the `mpc.gen` and `mpc.gencost` tables in the `case.mat` file that's created on the server, comparing against known-good scenarios: 378 was the final cost curve tuning scenario (with dummy generators scaled to zero), scenario 394 is our latest and greatest with all of the fixes. E.g., in MATLAB:
```
>> case_378 = load('PATH_TO_WHEREVER\scenario_378\case.mat');
>> case_394 = load('PATH_TO_WHEREVER\scenario_394\case.mat');
>> gencost_diff_7894 = case_394.mpc.gencost - case_378.mpc.gencost;
>> sum(abs(gencost_diff_7894),1)
ans =
         0         0         0         0    0.0000    0.0000    0.0053
>> gen_diff_7894 = case_394.mpc.gen - case_378.mpc.gen;
>> sum(abs(gen_diff_7894),1)

ans =
  Columns 1 through 8
         0         0         0         0         0    0.0000         0         0
  Columns 9 through 16
   67.4060    0.0000         0         0         0         0         0         0
  Columns 17 through 24
         0         0         0         0    0.1791         0         0         0
  Column 25
         0
```
The only differences are from rounding (`gencost` column 7, `gen` column 21) or from restoring the Pmax of dummy generators (`gen` column 9).
